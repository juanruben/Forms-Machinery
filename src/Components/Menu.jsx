import React from 'react';

function Menu() {
    return (
        <div>
            Men√∫
        </div>
    );
}

export default Menu;
