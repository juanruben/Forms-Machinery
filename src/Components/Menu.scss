.sidebar{
    width: 256px;

    .content{
        padding: 16px;
        height: 100%;

        .sidebarLink{
            display: block;
            text-decoration: none;
        }
    }
}
