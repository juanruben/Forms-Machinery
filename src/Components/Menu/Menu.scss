.sidebar {
  width: 256px;

  .content {
    padding: 16px;
    height: 100%;

    .sidebarLink {
      display: block;
      text-decoration: none;
    }

    button {
      background: none;
      border: none;
      padding: 0;
      text-decoration: none;
      cursor: pointer;
    }
  }
}
