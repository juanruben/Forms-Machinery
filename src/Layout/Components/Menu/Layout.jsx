import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { Collapse } from 'reactstrap';

export default class Layout extends Component {
  constructor(props) {
    super(props);
    this.state = {
      collapse: false,
    };
    this.toggle = this.toggle.bind(this);
  }

  toggle(e, tab) {
    const { collapse } = this.state;
    e.preventDefault();
    if (collapse !== tab) {
      this.setState({
        collapse: tab,
      });
    } else {
      this.setState({
        collapse: 0,
      });
    }
  }

  render() {
    const { items, path } = this.props;
    const { collapse } = this.state;
    return (
      // eslint-disable-next-line react/jsx-filename-extension
      <>
        {items.map((item, index) => (
          <Link to={path + item.link} onClick={(e) => { this.toggle(e, index + 1); }} key={index}>
            <li className="li-link">
              <span className="sidebar-label">
                {item.name}
              </span>
              <ul>
                {typeof item.submenu !== 'undefined' ? (
                  <Collapse isOpen={collapse === (index + 1)}>
                    {item.submenu.map((item2, index2) => (
                      <Link to={path + item2.link} key={index2}>
                        <li className="li-sub-link">
                          <span className="sidebar-label">
                            {item2.name}
                          </span>
                        </li>
                      </Link>
                    ))}
                  </Collapse>
                ) : null}
              </ul>
            </li>
          </Link>
        ))}
      </>
    );
  }
}
