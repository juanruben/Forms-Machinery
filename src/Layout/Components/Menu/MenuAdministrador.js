import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { Collapse } from 'reactstrap';

const menu = [
  {
    name: 'Registro Hist√≥rico',
    link: '/registro-historico/',
  },
  {
    name: 'Admin. Clientes',
    link: '#',
    submenu: [
      {
        name: 'Agregar Cliente',
        link: '/cliente/add',
      },
      {
        name: 'Admin. Clientes',
        link: '/cliente/lists',
      },
    ],
  },
  {
    name: 'Admin. Usuarios',
    link: '#',
    submenu: [
      {
        name: 'Agregar Usuario',
        link: '/usuario/add',
      },
      {
        name: 'Admin. Usuario',
        link: '/usuario/lists',
      },
    ],
  },
  {
    name: 'Admin. Maquina',
    link: '#',
    submenu: [
      {
        name: 'Agregar Maquina',
        link: '/maquina/add',
      },
      {
        name: 'Admin. Maquina',
        link: '/maquina/lists',
      },
    ],
  },
  {
    name: 'Admin. Formularios',
    link: '#',
    submenu: [
      {
        name: 'Agregar Formularios',
        link: '/formulario/add',
      },
      {
        name: 'Admin. Formularios',
        link: '/formulario/lists',
      },
    ],
  },
];


export default class MenuAdministrador extends Component {
  constructor(props) {
    super(props);
    this.state = {
      collapse: false,
      path: '/administrador',
    };
    this.toggle = this.toggle.bind(this);
  }

  toggle(e, tab) {
    const { collapse } = this.state;
    e.preventDefault();
    if (collapse !== tab) {
      this.setState({
        collapse: tab,
      });
    } else {
      this.setState({
        collapse: 0,
      });
    }
  }

  render() {
    const { collapse, path } = this.state;
    return (
      // eslint-disable-next-line react/jsx-filename-extension
      <>
        {menu.map((item, index) => (
          <Link to={path + item.link} onClick={(e) => { this.toggle(e, index + 1); }} key={index}>
            <li className="li-link">
              <span className="sidebar-label">
                {item.name}
              </span>
              <ul>
                {typeof item.submenu !== 'undefined' ? (
                  <Collapse isOpen={collapse === (index + 1)}>
                    {item.submenu.map((item2, index2) => (
                      <Link to={path + item2.link} key={index2}>
                        <li className="li-sub-link">
                          <span className="sidebar-label">
                            {item2.name}
                          </span>
                        </li>
                      </Link>
                    ))}
                  </Collapse>
                ) : null}
              </ul>
            </li>
          </Link>
        ))}
      </>
    );
  }
}
