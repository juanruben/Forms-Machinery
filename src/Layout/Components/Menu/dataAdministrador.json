[
    {
        "name": "Registro Hist√≥rico",
        "link": "/registro-historico/"
    },
    {
        "name": "Admin. Clientes",
        "link": "#",
        "submenu": [
            {
                "name": "Agregar Cliente",
                "link": "/cliente/add"
            },
            {
                "name": "Admin. Clientes",
                "link": "/cliente/lists"
            }
        ]
    },
    {
        "name": "Admin. Usuarios",
        "link": "#",
        "submenu": [
            {
                "name": "Agregar Usuario",
                "link": "/usuario/add"
            },
            {
                "name": "Admin. Usuario",
                "link": "/usuario/lists"
            }
        ]
    },
    {
        "name": "Admin. Maquina",
        "link": "#",
        "submenu": [
            {
                "name": "Agregar Maquina",
                "link": "/maquina/add"
            },
            {
                "name": "Admin. Maquina",
                "link": "/maquina/lists"
            }
        ]
    },
    {
        "name": "Admin. Formularios",
        "link": "#",
        "submenu": [
            {
                "name": "Agregar Formularios",
                "link": "/formulario/add"
            },
            {
                "name": "Admin. Formularios",
                "link": "/formulario/lists"
            }
        ]
    }
]