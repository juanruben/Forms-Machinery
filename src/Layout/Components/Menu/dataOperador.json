{
    "name": "Entradas y Salidas",
    "link": "#",
    "submenu": [
        {
            "name": "Check In",
            "link": "/cliente/agregar"
        },
        {
            "name": "Check Out",
            "link": "/cliente/lists"
        }
    ]
}