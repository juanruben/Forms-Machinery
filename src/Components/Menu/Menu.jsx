import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { useStateValue } from '../../State';
import Logo from '../Logo/Logo';
import Avatar from '../Avatar/Avatar';
import './Menu.scss';

const Menu = (props) => {
    const { items, username, avatar } = props;
    const state = useStateValue();

    const test = () => {
        const [, dispatch] = state;
        dispatch({
            type: 'EXIT',
        });
    };

    return (
        <div className="sidebar">
            <div className="content">
                <Logo />
                <Avatar title={username} image={avatar} />
                {items.map((item) => (
                    <Link key={item.id} to={item.path} className="sidebarLink">{item.title}</Link>
                ))}
                <button type="button" onClick={test}>Salir</button>
            </div>
        </div>
    );
};

Menu.propTypes = {
    items: PropTypes.arrayOf(
        PropTypes.shape({
            id: PropTypes.number.isRequired,
            title: PropTypes.string.isRequired,
            path: PropTypes.string.isRequired,
        }).isRequired,
    ).isRequired,
    username: PropTypes.string,
    avatar: PropTypes.string,
};

Menu.defaultProps = {
    username: '',
    avatar: '',
};

export default Menu;
