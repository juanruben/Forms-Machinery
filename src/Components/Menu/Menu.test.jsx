import React from 'react';
import { shallow } from 'enzyme';
import Menu from './Menu';

describe('Menu', () => {
    it('should render ok', () => {
        const wrapper = shallow(<Menu items={[]} />);
        expect(wrapper.debug()).toMatchSnapshot();
    });

    test('several items', () => {
        const items = [
            {
                id: 1,
                title: 'Dashboard',
                path: '/admin',
                icon: 'fas fa-user-tie',
            },
            {
                id: 2,
                title: 'Clientes',
                path: '/admin/clientes',
                icon: 'fas fa-user-tie',
            },
        ];
        const wrapper = shallow(<Menu items={items} />);
        expect(wrapper.find('NavLink')).toHaveLength(2);
    });

    test('one item', () => {
        const items = [
            {
                id: 1,
                title: 'Dashboard',
                path: '/admin',
                icon: 'fas fa-user-tie',
            },
        ];
        const wrapper = shallow(<Menu items={items} />);
        expect(wrapper.find('NavLink')).toHaveLength(1);
    });

    test('zero items', () => {
        const wrapper = shallow(<Menu items={[]} />);
        expect(wrapper.find('NavLink')).toHaveLength(0);
    });
});
