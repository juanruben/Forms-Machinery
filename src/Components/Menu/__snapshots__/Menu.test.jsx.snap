// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should render ok 1`] = `
"<div className=\\"sidebar\\">
  <div className=\\"content\\">
    <Logo padding={0} maxWidth={100} />
    <button type=\\"button\\" onClick={[Function: test]}>
      Salir
    </button>
  </div>
</div>"
`;
